src/lib/yang/data.lua

```diff
@@ -715,14 +716,28 @@ function data_printer_from_grammar(production)
       file:write(' ')
    end
    local function body_printer(productions, order)
+      -- Iterate over productions trying to translate to other statements. This
+      -- is used for example in choice statements raising the lower statements
+      -- in case blocks up to the level of the choice, in place of the choice.
+      for keyword,production in pairs(productions) do
+         local translator = translators[production.type]
+         if translator ~= nil then
+            local statements = translator(keyword, production)
+            for k,v in pairs(statements) do productions[k] = v end
```

Comment:
Apparently, it's not OK to add new keys to a table while iterating over it http://lua-users.org/lists/lua-l/2005-09/msg00512.html There was a bug recently related with that https://github.com/Igalia/snabb/pull/831 After reworking the code and not directly adding the new elements to the iterated table, the issue got solved.

Maybe this code is working OK as it is now, but since it was tough to figuring out the why #831 was failing, I'd suggest to rework the code in some way to avoid modifying the table while looping it. Removing keys from a table while iterating it seems to be OK though.